src/sync/sync.nim

```diff
@@ -0,0 +1,238 @@
+                        logDetailed(&"[skip] {slug}: metadata is up-to-date")
+                      else:
+                        logNormal(&"[warn] {slug}: metadata is unsynced")
+                        seenUnsynced.incl skMetadata
+                        if conf.action.update:
+                          j[key].str = upstreamVal.stringVal
+                          result.add MetadataPair(
+                            trackJsonPath: trackExerciseConfigPath,
+                            updatedJson: j)
+                else:
+                  seenUnsynced.incl skMetadata
+          else:
+            logNormal(&"[error] {slug}: {metadataFilename} is missing")
+            seenUnsynced.incl skMetadata
+      else:
+        logDetailed(&"[skip] {slug}: does not exist in problem-specifications")
+    else:
+      logNormal(&"[error] {slug}: .meta dir missing")
+      seenUnsynced.incl skMetadata
+
+proc checkTests(exercises: seq[Exercise], seenUnsynced: var set[SyncKind]) =
+  for exercise in exercises:
+    let numMissing = exercise.tests.missing.len
+    let wording = if numMissing == 1: "test case" else: "test cases"
+
+    case exercise.status()
+    of exOutOfSync:
+      seenUnsynced.incl skTests
+      logNormal(&"[warn] {exercise.slug}: missing {numMissing} {wording}")
+      for testCase in exercise.testCases:
+        if testCase.uuid in exercise.tests.missing:
+          logNormal(&"       - {testCase.description} ({testCase.uuid})")
+    of exInSync:
+      logDetailed(&"[skip] {exercise.slug}: up-to-date")
+    of exNoCanonicalData:
+      logDetailed(&"[skip] {exercise.slug}: does not have canonical data")
+
+proc explain(syncKind: SyncKind): string =
+  case syncKind
+  of skDocs: "have unsynced docs"
+  of skFilepaths: "have unsynced filepaths"
+  of skMetadata: "have unsynced metadata"
+  of skTests: "are missing test cases"
+
+proc userSaysYes(noun: string): bool =
+  stderr.write &"sync the above {noun} ([y]es/[n]o)? "
+  let resp = stdin.readLine().toLowerAscii()
+  if resp == "y" or resp == "yes":
+    result = true
+
+proc sync*(conf: Conf) =
+  logNormal("Checking exercises...")
+
+  let probSpecsDir = initProbSpecsDir(conf)
+  var seenUnsynced: set[SyncKind]
+
+  try:
+    let exercises = toSeq findExercises(conf, probSpecsDir)
+    let psExercisesDir = probSpecsDir / "exercises"
+    let trackPracticeExercisesDir = joinPath(conf.trackDir, "exercises", "practice")
+
+    if skDocs in conf.action.scope:
+      let sdPairs = checkDocs(exercises, psExercisesDir,
+                              trackPracticeExercisesDir, seenUnsynced, conf)
+      if sdPairs.len > 0:
+        if conf.action.update:
+          if conf.action.yes or userSaysYes("docs"):
+            for sdPair in sdPairs:
+              # TODO: don't replace first top-level header?
+              # For example: the below currently writes `# Description`
+              # instead of `# Instructions`
```

Comment:
This should be fixed once we update the `description.md` documents to `instructions.md` documents. You could, alternatively, simply replaced `# Description` with `# Instructions` for the moment 🤷

Reply:
If I make a PR in `problem-specifications` that renames every `description.md` to `instructions.md`, is there anything that blocks us from merging it?

Reply:
Yes, because we want to not only rename to `instructions.md`, but while we're doing that, also split the `description.md` files into separate `instructions.md` and `introduction.md` files (if possible). Not having these files renamed gives us an easy way to identify which exercises haven't yet been converted. Note that this is still future work and not something we're focusing on right now.

For now, I think it's fine to just have the `# Description` heading (or do a string replace to `# Instructions`).

Reply:
Let's replace with `# Instructions`.